### PR TITLE
Added: GitHub workflows to automatically test the Code Examples in the SDK documentation

### PR DESCRIPTION
**Pull request by iftwigs**

### CODE REVIEW

File: docs/dvui/explanations.md [hunk 1]
```diff
@@ -7,26 +7,43 @@
 
 There are two options to manage users in the Document Validation UI:
 
-- The first option is to [create a single user](https://app.konfuzio.com/accounts/signup/) with limited permissions (i.e. with the [Reader role](https://help.konfuzio.com/modules/superuserroles/index.html)) and [user token](https://app.konfuzio.com/v3/swagger/#/auth/auth_create), that anyone who interacts with the user interface can use in a project. Since the token will be public, this option is best suited for a read-only view.
-- The second one is to create as many users as needed, each with their own token. This is best suited for projects that use [single sign-on authentication](https://dev.konfuzio.com/web/api-v3.html#single-sign-on-sso-authentication) and/or where users will be interacting with the Documents (annotating and revising), since it allows for fine-grained control over who can access what and it allows the application to keep a record of the different actions performed by them.
+- The first option is to [create a single user](https://app.konfuzio.com/accounts/signup/) with limited permissions
```
Comment (da-konfuzio) [inline docs/dvui/explanations.md:10]: It looks like this document was automatically reformatted by mistake

File: docs/dvui/guides.md [hunk 2]
```diff
@@ -5,27 +5,32 @@
 
 ## How to integrate the Document Validation UI into custom solutions?
```
Comment (da-konfuzio) [inline docs/dvui/guides.md:6]: It looks like this document was automatically reformatted by mistake

**Comment by da-konfuzio** [comment]
Was this added by mistake? docs/_static/img/bboxes_characters.png

File: docs/sdk/boilerplates/run_doctesting.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+"""Run doctests on modules."""
+import os
+
+doctest_file_paths = ['konfuzio_sdk/trainer/file_splitting.py']
+
+for path in doctest_file_paths:
+    os.system(f'python -m doctest -v {path}')
```
Comment (da-konfuzio) [inline docs/sdk/boilerplates/run_doctesting.py:7]: I realized that vecause `os.system` executes a subshell, no exceptions will be raised if the command fails or if one of the tested docstrings fails. Let's rather keep it simple and have the full list in `python-app.yml` :slightly_smiling_face:

**Comment by iftwigs** [comment]
> Was this added by mistake? docs/_static/img/bboxes_characters.png
no, my initial idea was to replace an illustrative screenshot with the new one because I was running the code example with the different document, but then it became clear that it is not visible to the external user of the documentation so the old one is still okay.
> Same with the json files
these are mistakenly removed, will return

**Review by da-konfuzio — APPROVED** [review]

**Review by da-konfuzio — APPROVED** [review]

**Pull request merged**